### PR TITLE
Support specifying workload identity at deployment create time

### PR DESCRIPTION
**Pull request by jlaneve**
## Description

This PR adds support for a new field, Workload Identity, on deployment create time. The corresponding API change has been merged and goes out next week.

There is a change in behavior for deployment yaml files / deployments as code:
- the deployment yaml file has two sections: config (can be edited) and metadata (can't be edited). so this PR moves workload ID from metadata to config
-  if I have it set up today without the workload ID field in config section, it still works. so if a user wants to start using this workload ID with a pre-existing yaml file, they just need to move the field themselves

## 🎟 Issue(s)

n/a

## 🧪 Functional Testing

Built the CLI locally and ran against dev.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [x] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ build:
 
 core_api_gen:
     ifeq (, $(shell which oapi-codegen))
-	go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@latest
+	go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@v1.12.4
```
Comment (jlaneve) [inline Makefile:42]: note that if this isn't pinned, a new developer will have a newer version installed which causes minor formatting differences across all generated files

File: cloud/deployment/deployment_test.go [hunk 2]
```diff
@@ -901,15 +901,15 @@ func TestCreate(t *testing.T) {
 		defer testUtil.MockUserInput(t, "1")()
 
 		// Call the Create function
-		err := Create("test-name", ws, "test-desc", csID, "4.2.5", dagDeploy, CeleryExecutor, "", "", SmallScheduler, "", "", "", "", "", "", astroplatformcore.DeploymentTypeHYBRID, 0, 0, mockPlatformCoreClient, mockCoreClient, false)
+		err := Create("test-name", ws, "test-desc", csID, "4.2.5", dagDeploy, CeleryExecutor, "", "", SmallScheduler, "", "", "", "", "", "", "", astroplatformcore.DeploymentTypeHYBRID, 0, 0, mockPlatformCoreClient, mockCoreClient, false)
```
[the remark this answers is not in the document]
Comment (jlaneve) [reply cloud/deployment/deployment_test.go:904]: ah yep good catch, done

File: Makefile (hunk 1 above)
Comment (kushalmalani) [reply Makefile:42]: We are actually planning to move to v2 of this library in a separate PR. Just FYI

Comment (feluelle) [reply Makefile:42]: Hey, I have created https://github.com/astronomer/astro-cli/pull/1552, hopefully this helps new devs to always use the correct version.

**Comment by sunkickr** [comment]
should we add a `--workload-identity` flag to `astro deployment create/update`?

**Review by sunkickr — APPROVED** [review]
LGTM

**Pull request merged**